src/report.py

```diff
@@ -0,0 +1,87 @@
+import os
+import utils
+from buildercore import core, project, utils as core_utils
+from functools import wraps
+
+def print_list(row_list, checkboxes=True):
+    "given a list of things, prints a markdown list to stdout with optional checkboxes"
+    template = "- %s"
+    if checkboxes:
+        template = "- [ ] %s"
+    for row in row_list:
+        print(template % row)
+
+def report(fn):
+    "wraps a task, printing it's results as a sorted list to stdout"
+    @wraps(fn)
+    def _wrapped(checkboxes=True, ordered=True, *args, **kwargs):
+        ordered = utils.strtobool(ordered)
+        checkboxes = utils.strtobool(checkboxes)
+        results = fn(*args, **kwargs)
+        if not results:
+            return
+        # TODO: this could be better. ci first, continuumtest next, etc
+        if ordered:
+            results.sort()
+        print_list(results, checkboxes)
+        return results
+    return _wrapped
+
+@report
+def all_projects():
+    "returns a list of all project names"
+    return project.project_list()
+
+@report
+def all_formulas():
+    "returns a list of all known formulas"
+    formula_url_list = filter(None, list(set(core_utils.shallow_flatten(project.project_formulas().values()))))
+    formula_list = map(lambda url: os.path.basename(url), formula_url_list)
+    return formula_list
+
+@report
+def all_ec2_projects():
+    "return a list of all project names whose projects have a truthy ec2 section (eg, not {}, None or False)"
+    def has_ec2(pname, pdata):
+        if pdata.get('aws') and pdata['aws'].get('ec2'):
+            return pname
+        for alt_name, alt_data in pdata.get('aws-alt', {}).items():
+            if alt_data.get('ec2'):
+                return pname
+    results = [has_ec2(pname, pdata) for pname, pdata in project.project_map().items()]
+    results = filter(None, results)
+    return results
+
+
+@report
+def all_ec2_instances(state=None):
+    "all ec2 instances. set `state` to `running` to see all running ec2 instances"
```

Comment:
returns a list of all ec2 instance names. set `state` to `running` to see all running ec2 instances